includes/widget/class-urlslab-widget-cache.php

```diff
@@ -166,14 +168,29 @@ public function init_wp_admin_menu( string $plugin_name, WP_Admin_Bar $wp_admin_
 	}
 
 	private function is_cache_enabled(): bool {
-		return ! is_admin() && ! is_user_logged_in() && isset( $_SERVER['REQUEST_URI'] ) && ( ! is_404() || $this->get_option( self::SETTING_NAME_CACHE_404 ) ) && $this->get_option( self::SETTING_NAME_PAGE_CACHING );
-	}
+		if ( ! isset( $_SERVER['REQUEST_METHOD'] ) || 'GET' !== $_SERVER['REQUEST_METHOD'] ) {
+			return false;
+		}
 
-	private function start_rule(): bool {
-		if ( ! isset( $_SERVER['REQUEST_URI'] ) ) {
+		if ( isset( $_SERVER['HTTP_COOKIE'] ) && preg_match( '/(comment_author|wp-postpass|logged|wptouch_switch_toggle)/', $_SERVER['HTTP_COOKIE'] ) ) {
```

Comment:
we should sanitize this variables using `sanitize_text_field()` like other places in the code

Reply:
???? why we don't use the value at all, it is used just for check, value is not used in the system.